### PR TITLE
Add a re-calculate column width button

### PR DESCRIPTION
**Pull request by deeptailor**
Since new telemetry tables were released, the auto-size functionality was gotten rid of. This re-implements it, via a user button for performance reasons. 

## Author Checklist
Changes address original issue? Y
Unit tests included and/or updated with changes? N/A
Command line build passes? Y
Changes have been smoke-tested? Y

### CODE REVIEW

**Comment by deeptailor** [comment]
@charlesh88 - Take a look at the icon and I discovered one issue, the separator is collapsing with a zero height.

File: src/plugins/telemetryTable/components/table.vue [hunk 1]
```diff
@@ -462,21 +473,34 @@ export default {
...
-                totalWidth += columnWidth;
+                sizingTableChildren.forEach(sizingRowEl => {
+                    let sizingCells = Array.from(sizingRowEl.children),
+                        cell = sizingCells[headerIndex];
+
+                    if (columnWidths[headerKey]) {
+                        let currentWidth = columnWidths[headerKey],
+                            newWidth = cell.offsetWidth;
+
+                        if (newWidth > currentWidth) {
```
Comment (shefalijoshi) [inline src/plugins/telemetryTable/components/table.vue]: What happens if the newWidth is shorter than the current width? Do we only want to track increasing widths?

File: src/plugins/telemetryTable/components/table.vue [hunk 2]
```diff
@@ -462,21 +473,34 @@ export default {
...
-            let sizingCells = Array.from(sizingRowEl.children);
             let headerKeys = Object.keys(this.headers);
+            let sizingTableChildren = Array.from(this.sizingTable.children);
 
             headerKeys.forEach((headerKey, headerIndex)=>{
-                let cell = sizingCells[headerIndex];
-                let columnWidth = cell.offsetWidth;
-                columnWidths[headerKey] = columnWidth;
-                totalWidth += columnWidth;
+                sizingTableChildren.forEach(sizingRowEl => {
```
Comment (shefalijoshi) [inline src/plugins/telemetryTable/components/table.vue]: Recommend not having nested loops.

Comment (deeptailor) [reply src/plugins/telemetryTable/components/table.vue]: Simplified it, does not need any nested loops anymore

File: src/plugins/telemetryTable/components/table.vue (hunk 1 above)
Comment (deeptailor) [reply src/plugins/telemetryTable/components/table.vue]: Simplified it, does not need this if block anymore

**Comment by deeptailor** [comment]
@shefali - I simplified the code to remove the questionable logic. Please review and merge

File: src/plugins/telemetryTable/components/table.vue [hunk 3]
```diff
@@ -461,23 +484,28 @@ export default {
...
+                }
+                totalWidth += columnWidths[headerKey];
             });
 
             this.columnWidths = columnWidths;
             this.totalWidth = totalWidth;
 
             this.calculateScrollbarWidth();
+
+            return Promise.resolve();
```
Comment (psarram) [inline src/plugins/telemetryTable/components/table.vue]: Why would this function need to return a promise? It seems unnecessary.

**Comment by psarram** [comment]
@deeptailor I have one comment, otherwise looks great.

File: src/plugins/telemetryTable/components/table.vue (hunk 3 above)
Comment (deeptailor) [reply src/plugins/telemetryTable/components/table.vue]: thanks for catching this, I was using it to clear out the sizing table systematically. But i simplified it and reduced the complication

**Comment by deeptailor** [comment]
@psarram - pushed a small change simplifying and removing the promise. Please re-review and merge

**Comment by psarram** [comment]
**Reviewer Checklist**

- Changes appear to address issue? Y
- Appropriate unit tests included? N
- Code style and in-line documentation are appropriate? Y
- Commit messages meet standards? Y

**Pull request merged**